### PR TITLE
GoVetBearTest.py: Change go test files

### LINKED ISSUE #2337: GoVetBear throwing error on appveyor
GoVetBearTest failing with error `The local bear 'GoVetBear' yields a result although it shouldn't.`
Link: https://ci.appveyor.com/project/coala/coala-bears/build/1.0.9263/job/4xamw8qmesan9aat

### PR DESCRIPTION
**Pull request by MacBox7**
The previous test files had invalid syntax and thus
resulting into failure

Fixes #2337

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by MacBox7** [comment]
@manankalra `vet` checks for `semantic` errors and in here both the files are `syntactically` incorrect. The purpose of `vet_bad.go` is to generate `semantic` error and not `syntax` error. Hope you understand.

**Comment by jayvdb** [comment]
@MacBox7 , it is still useful to give `go vet` syntax errors to ensure it finds and reports an error correctly.

You can add lots of tests cases; that is better than few test cases.

More analysis needed.  Specifically why were our tests passing yesterday, and are failing now.  What changed?  Please explain that on the issue https://github.com/coala/coala-bears/issues/2337  Also, why are the tests passing on Travis/Circle, and not Appveyor?  Likely it is related to a go/govet version change, and caching on the VMs.

**Comment by MacBox7** [comment]
@jayvdb added seperate test cases for checking `syntax` and `semantic` errors. Is there anything else that needs to be taken care of?

**Comment by MacBox7** [comment]
We can install the stable version from https://golang.org/dl/ but it would pin our go version to 1.10. Shouldn't the version update be handled in a different PR? I don't think installing from ppa would be a good idea still there is no official ppa that go lang supports. You can check this out https://github.com/golang/go/wiki/Ubuntu they clearly mention that they don't support it.

**Comment by virresh** [comment]
Since the current method of go is also from installing ppa's, and the issue is due to golang's version update, I thought PPA update would fit it in the PR fixing the bug, although yes probably we can have a separate issue for that :+1:

File: tests/go/test_files/vet_good.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 package main
 
 func main() {
-	return 1
+
```
Comment (virresh) [inline tests/go/test_files/vet_good.go]: Probably consider placing the sample hello world program of go in here
Better than a blank function :sweat_smile: 
```
package main
import "fmt"
func main() {
    fmt.Printf("%q","hello world")
}
```

Rest looks good to go

Comment (MacBox7) [reply tests/go/test_files/vet_good.go]: That's an unnecessary request, don't you think :wink:.

Comment (virresh) [reply tests/go/test_files/vet_good.go]: Just a suggestion :smiley: 
My two cents for making coala better :)

**Comment by anctartica** [comment]
Change the commit message using `git commit --amend` and then push again

File: tests/go/GoVetBearTest.py [hunk 2]
```diff
@@ -14,13 +14,19 @@ class GoVetBearTest(LocalBearTestHelper):
...
-    def test_run(self):
-        self.check_validity(self.uut, [], self.good_file)
-        self.check_invalidity(self.uut, [], self.bad_file)
+    def test_syntax_error(self):
+        good_file = os.path.join(os.path.dirname(__file__),
+                                 'test_files',
+                                 'vet_good.go')
+        bad_file = os.path.join(os.path.dirname(__file__),
+                                'test_files',
+                                'vet_bad_semantics.go')
```
Comment (sangamcse) [inline tests/go/GoVetBearTest.py]: In my opinion, it's a bad practice of writing test. Why not using `setUp` to set abs path of test_files and then call it in each test?
I mean you can do below code in setUp
`self.test_path = os.path.abspath(os.path.join(os.path.dirname(__file__), 'test_files'))`
And call it in each test by using `good_file = os.path.join(self.test_path, 'vet_good.go')`.

Comment (MacBox7) [reply tests/go/GoVetBearTest.py]: Yes, I felt the same. But followed the same practice which was used before. Would update it soon :smile;

File: tests/go/test_files/vet_good.go (hunk 1 above)
Comment (jayvdb) [reply tests/go/test_files/vet_good.go]: It is a reasonable request, given that our previous example good go file wasnt very good.
It is possible that a future version of go disallows an empty function.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala-bears/pull/2338/commits/f8b779f1c33e4db41e184dc2b9df1ee5e969698c

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**